query-connector/src/app/backend/usergroup-management.ts

```diff
@@ -1,65 +1,200 @@
+    const existingGroups = await getAllUserGroups();
+    const groupExists =
+      existingGroups.items?.some((group) => {
+        return group.name === newName;
+      }) ?? false;
+
+    if (groupExists) {
+      console.warn(`Group with name '${newName}' already exists.`);
+      return `Group '${newName}' already exists.`;
+    }
+
+    const updateUserGroupMembersQuery = `
+      UPDATE usergroup
+      SET name = $1
+      WHERE id = $2
+      RETURNING id, name;
+    `;
+
+    const result = await dbClient.query(updateUserGroupMembersQuery, [
+      newName,
+      id,
+    ]);
+
+    if (result.rows.length === 0) {
+      throw new Error(`User group with ID '${id}' not found.`);
+    }
+
+    // Get updated memberSize and querySize from getUserGroup
+    const userGroups = await getAllUserGroups();
+    const updatedGroup =
+      userGroups?.items?.find((group) => group.id === id) ?? null;
+
+    return {
+      id: result.rows[0].id,
+      name: result.rows[0].name,
+      member_size: updatedGroup?.member_size ?? 0,
+      query_size: updatedGroup?.query_size ?? 0,
+    };
+  } catch (error) {
+    console.error("Error updating user group:", error);
+    throw error;
+  }
+}
 
-  return result.rows.map((row) => ({
-    id: row.id,
-    username: row.username,
-    first_name: row.first_name,
-    last_name: row.last_name,
-    qc_role: row.qc_role,
-    userGroupMemberships: [
-      {
-        id: row.membership_id,
-        usergroup_id: groupId,
-        group_name: row.group_name,
-        is_member: row.is_member,
-      },
-    ],
-  }));
+/**
+ * Deletes a user group by its unique identifier.
+ * @param id - The unique identifier of the user group to delete.
+ * @returns The deleted user group or an error if the deletion fails.
+ */
+export async function deleteUserGroup(id: string): Promise<UserGroup | string> {
+  if (!(await superAdminAccessCheck())) {
+    throw new Error("Unauthorized");
+  }
+
+  try {
+    await dbClient.query(
+      "DELETE FROM usergroup_to_users WHERE usergroup_id = $1",
+      [id],
+    );
+    await dbClient.query(
+      "DELETE FROM usergroup_to_query WHERE usergroup_id = $1",
+      [id],
+    );
+    const deleteGroupQuery = `
+      DELETE FROM usergroup
+      WHERE id = $1
+      RETURNING id, name;
+    `;
+
+    const result = await dbClient.query(deleteGroupQuery, [id]);
+
+    if (result.rows.length === 0) {
+      throw new Error(`User group with ID '${id}' not found.`);
+    }
+
+    return {
+      id: result.rows[0].id,
+      name: result.rows[0].name,
+      member_size: 0,
+      query_size: 0,
+    };
+  } catch (error) {
+    console.error("Error deleting user group:", error);
+    throw error;
+  }
+}
+
+/**
+ * Adds users to a user group.
+ * @param groupId - The unique identifier of the user group.
+ * @param userId - The unique identifiers of the users to add.
+ * @returns The user IDs of the users added to the group.
+ */
+export async function addSingleUserToGroup(
```

Comment:
suggestion: instead of having multiple single vs multiple methods here, is there a way we might be able to [overload](https://www.typescriptlang.org/docs/handbook/2/functions.html#overload-signatures-and-the-implementation-signature) the method here to take both a `userId` and `userId[]`, and then based on the inputted type, stitch the insertion SQL accordingly? That way we don't have to maintain multiple CRUD interfaces to the same underlying database functionality